staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -131,6 +131,10 @@ func (IntOrString) OpenAPISchemaType() []string { return []string{"string"} }
 // the OpenAPI spec of this type.
 func (IntOrString) OpenAPISchemaFormat() string { return "int-or-string" }
 
+// OpenAPIV3OneOfTypes is used by the kube-openapi generator when constructing
+// the OpenAPI v3 spec of this type.
+func (IntOrString) OpenAPIV3OneOfTypes() []string { return []string{"int", "string"} }
```

Comment:
is "int" a valid type? I thought it had to be "integer" - https://swagger.io/docs/specification/data-models/data-types/#numbers

if so, I'm surprised nothing caught this

Reply:
Yeah, I'm pretty sure we've set-up a script to validate the output of OpenAPI v3 so that we never generate invalid OpenAPI in `update-openapi-spec.sh`, but I can't see it. @Jefftree. Additionally, we could serialize to protobuf since that seems to add its own validation too.

Reply:
Thanks for pointing this out. I've pushed an update.